sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -437,8 +437,11 @@ export abstract class LinkEntity<LinkT extends Receiver | AwaitableSender | Requ
     if (this._context.tokenCredential instanceof SharedKeyCredential) {
       tokenObject = this._context.tokenCredential.getToken(this.audience);
       tokenType = TokenType.CbsTokenTypeSas;
-      // renew sas token in every 45 minutess
-      this._tokenTimeout = (3600 - 900) * 1000;
+
+      if (this._context.tokenCredential.renewable) {
```

Comment:
@ramya-rao-a , @HarshaNalluru  - is this worth it? If we didn't care then we can just let a useless token renewal timer run but we wouldn't need an API extension in core-amqp.

We could also do the same here by checking the type of SharedKeyCredential but that seems less clean to me.

Reply:
This seems fine to me.

Should this affect event-hubs too?

Reply:
It will, I'm just starting with service bus but we'll have to plumb the same code through there as well.

(odds are _excellent_ that it looks similar there)

Reply:
It was simpler to just say that if an AccessToken returns 0 we just don't activate token renewal.